src/ray/gcs/redis_async_context.h

```diff
@@ -92,8 +104,34 @@ class RedisAsyncContext {
   /// should be minimum.
   std::mutex mutex_;
```

Comment:
Actually I'm not quite sure we should use mutex and multi-threading here.
The only operation which might involve multi-threaded operation is `operate` invoked from another thread, all other operations are posted into `io_service`; curious why do we need concurrent `operate` calls?

Reply:
Actually, I don't know either. But I suspect that the predecessors decided to use a mutex because they saw this: https://github.com/redis/hiredis/blob/8d8703ee6149ffd3b38cebedd71ad8d1fac63cc7/README.md?plain=1#L152.

Reply:
They want to make `redisAsyncContext` thread-safe.